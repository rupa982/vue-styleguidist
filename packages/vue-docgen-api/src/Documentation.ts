import Map from 'ts-map'

export type BlockTag = ParamTag | Tag

export interface Module {
	name: string
	path: string
}

/**
 * Universal model to display origin
 */
export interface Descriptor {
	extends?: Module
	mixin?: Module
}

export interface ParamType {
	name: string
	elements?: ParamType[]
}

export interface UnnamedParam {
	type?: ParamType
	description?: string | boolean
}

export interface Param extends UnnamedParam {
	name?: string
}

interface RootTag {
	title: string
}

export interface Tag extends RootTag {
	content: string | boolean
}

export interface ParamTag extends RootTag, Param {}

export interface DocBlockTags {
	description?: string
	tags?: Array<ParamTag | Tag>
}

interface EventType {
	names: string[]
}

interface EventProperty {
	type: EventType
	name?: string
	description?: string | boolean
}

export interface EventDescriptor extends DocBlockTags, Descriptor {
	name: string
	type?: EventType
	properties?: EventProperty[]
}

export interface PropDescriptor extends Descriptor {
	type?: { name: string; func?: boolean }
	description?: string
	required?: boolean
	defaultValue?: { value: string; func?: boolean }
	tags?: { [title: string]: BlockTag[] }
	values?: string[]
	name: string
}

export interface MethodDescriptor extends Descriptor {
	name: string
	description?: string
	returns?: UnnamedParam
	tags?: { [key: string]: BlockTag[] }
	params?: Param[]
	modifiers?: string[]
	[key: string]: any
}

export interface SlotDescriptor extends Descriptor {
	name: string
	description?: string
	bindings?: ParamTag[]
	scoped?: boolean
	tags?: { [key: string]: BlockTag[] }
}

export interface ComponentDoc {
	displayName: string
	exportName: string
	description?: string
	props?: PropDescriptor[]
	methods?: MethodDescriptor[]
	slots?: SlotDescriptor[]
	events?: EventDescriptor[]
	tags?: { [key: string]: BlockTag[] }
	docsBlocks?: string[]
	[key: string]: any
}

export default class Documentation {
	private propsMap: Map<string, PropDescriptor>
	private methodsMap: Map<string, MethodDescriptor>
	private slotsMap: Map<string, SlotDescriptor>
	private eventsMap: Map<string, any>
	private dataMap: Map<string, any>
	private docsBlocks: string[] | undefined
	private originExtendsMixin: Descriptor
	public readonly componentFullfilePath: string

	constructor(fullFilePath: string) {
		this.componentFullfilePath = fullFilePath
		this.propsMap = new Map()
		this.methodsMap = new Map()
		this.slotsMap = new Map()
		this.eventsMap = new Map()
		this.originExtendsMixin = {}

		this.dataMap = new Map()
	}

	public setOrigin(origin: Descriptor) {
		this.originExtendsMixin = origin.extends ? { extends: origin.extends } : {}

		this.originExtendsMixin = origin.mixin ? { mixin: origin.mixin } : {}
	}

	public setDocsBlocks(docsBlocks: string[]) {
		this.docsBlocks = docsBlocks
	}

	public set(key: string, value: any) {
		this.dataMap.set(key, value)
	}

	public get(key: string): any {
		return this.dataMap.get(key)
	}

	public getPropDescriptor(propName: string): PropDescriptor {
		const vModelDescriptor = this.propsMap.get('v-model')
		return vModelDescriptor && vModelDescriptor.name === propName
			? vModelDescriptor
			: this.getDescriptor(propName, this.propsMap, () => ({
					name: propName
			  }))
	}

	public getMethodDescriptor(methodName: string): MethodDescriptor {
		return this.getDescriptor(methodName, this.methodsMap, () => ({
			name: methodName
		}))
	}

	public getEventDescriptor(eventName: string): EventDescriptor {
		return this.getDescriptor(eventName, this.eventsMap, () => ({
			name: eventName
		}))
	}

	public getSlotDescriptor(slotName: string): SlotDescriptor {
		return this.getDescriptor(slotName, this.slotsMap, () => ({
			name: slotName
		}))
	}

	public toObject(): ComponentDoc {
		const props = this.getObjectFromDescriptor(this.propsMap)
		const methods = this.getObjectFromDescriptor(this.methodsMap)
		const events = this.getObjectFromDescriptor(this.eventsMap)
		const slots = this.getObjectFromDescriptor(this.slotsMap)

		const obj: { [key: string]: any } = {}
		this.dataMap.forEach((value, key) => {
			if (key) {
				obj[key] = value
			}
		})

		if (this.docsBlocks) {
			obj.docsBlocks = this.docsBlocks
		}

		return {
			...obj,
			// initialize non null params
			description: obj.description || '',
			tags: obj.tags || {},

			// set all the static properties
			exportName: obj.exportName,
			displayName: obj.displayName,
			props,
			events,
			methods,
			slots
		}
	}

	private getDescriptor<T>(name: string, map: Map<string, T>, init: () => T): T {
		let descriptor = map.get(name)
		if (!descriptor) {
			descriptor = init()
			descriptor = { ...descriptor, ...this.originExtendsMixin }
			map.set(name, descriptor)
		}
		return descriptor
	}

	private getObjectFromDescriptor<T>(map: Map<string, T>): T[] | undefined {
		if (map.size > 0) {
			const obj: T[] = []
			map.forEach((descriptor, name) => {
				if (name && descriptor) {
					obj.push(descriptor)
				}
			})
			return obj
		} else {
			return undefined
		}
	}
}
