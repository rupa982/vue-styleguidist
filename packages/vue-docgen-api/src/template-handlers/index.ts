import propHandler from './propHandler'
import slotHandler from './slotHandler'
import eventHandler from './eventHandler'

export default [slotHandler, propHandler, eventHandler]
