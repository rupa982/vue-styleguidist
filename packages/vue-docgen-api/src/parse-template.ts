import * as pug from 'pug'
import { parse, TemplateChildNode, RootNode, BaseElementNode } from '@vue/compiler-dom'
import { SFCTemplateBlock } from '@vue/compiler-sfc'
import Documentation from './Documentation'
import cacher from './utils/cacher'

export interface TemplateParserOptions {
	functional: boolean
	rootLeadingComment: string[]
}

export type Handler = (
	documentation: Documentation,
	templateAst: TemplateChildNode,
	options: TemplateParserOptions
) => void

export default function parseTemplate(
	tpl: Pick<SFCTemplateBlock, 'content' | 'attrs'>,
	documentation: Documentation,
	handlers: Handler[],
	filePath: string
) {
	if (tpl && tpl.content) {
		const source =
			tpl.attrs && tpl.attrs.lang === 'pug'
				? pug.render(tpl.content, { filename: filePath })
				: tpl.content

		const ast: RootNode = cacher(() => parse(source), source)

		const functional = !!tpl.attrs.functional
		if (functional) {
			documentation.set('functional', functional)
		}

		const rootLeadingComment = extractRootLeadingComments(source)

		if (ast) {
			ast.children.forEach(child =>
				traverse(child, documentation, handlers, {
					functional,
					rootLeadingComment
				})
			)
		}
	}
}

function hasChildren(child: any): child is BaseElementNode {
	return !!child.children
}

export function traverse(
	templateAst: TemplateChildNode,
	documentation: Documentation,
	handlers: Handler[],
	options: TemplateParserOptions
) {
	const traverseAstChildren = (templateAst: TemplateChildNode) => {
		if (hasChildren(templateAst)) {
			const { children } = templateAst
			for (const childNode of children) {
				if (hasChildren(childNode)) {
					traverse(childNode, documentation, handlers, options)
				}
			}
		}
	}

	if (templateAst.type === 1) {
		handlers.forEach(handler => {
			handler(documentation, templateAst, options)
		})
	}

	traverseAstChildren(templateAst)
}

function extractRootLeadingComments(template: string): string[] {
	let t = template.trim()
	const comments = []
	while (/^<!--/.test(t)) {
		const endOfRootLeadingComment = t.indexOf('-->')
		const rootLeadingComment = t.slice(4, endOfRootLeadingComment).trim()
		comments.push(rootLeadingComment)
		t = t.slice(endOfRootLeadingComment + 3).trim()
	}

	return comments
}
