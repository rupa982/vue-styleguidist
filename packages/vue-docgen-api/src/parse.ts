import { readFile } from 'fs'
import * as path from 'path'
import { promisify } from 'util'
import { parseComponent } from 'vue-template-compiler'
import Documentation, { Descriptor } from './Documentation'
import parseScript, { Handler as ScriptHandler } from './parse-script'
import parseTemplate, { Handler as TemplateHandler } from './parse-template'
import scriptHandlers, { preHandlers } from './script-handlers'
import templateHandlers from './template-handlers'
import cacher from './utils/cacher'

const read = promisify(readFile)

const ERROR_EMPTY_DOCUMENT = 'The passed source is empty'

export { ScriptHandler, TemplateHandler }

export interface ParseOptions extends DocGenOptions, Descriptor {
	filePath: string
	/**
	 * In what language is the component written
	 * @default undefined - let the system decide
	 */
	lang?: 'ts' | 'js'
}

export interface DocGenOptions {
	/**
	 * Which exported variables should be looked at
	 * @default undefined - means treat all dependencies
	 */
	nameFilter?: string[]
	/**
	 * What alias should be replaced in requires and imports
	 */
	alias?: { [alias: string]: string }
	/**
	 * What directories should be searched when resolving modules
	 */
	modules?: string[]
	/**
	 * Handlers that will be added at the end of the script analysis
	 */
	addScriptHandlers?: ScriptHandler[]
	/**
	 * Handlers that will be added at the end of the template analysis
	 */
	addTemplateHandlers?: TemplateHandler[]
	/**
	 * Does parsed components use jsx?
	 * @default true - if you do not disable it, babel will fail with `(<any>window).$`
	 */
	jsx?: boolean
}

/**
 * parses the source and returns the doc
 * @param {string} source code whose documentation is parsed
 * @param {string} filePath path of the current file against whom to resolve the mixins
 * @returns {object} documentation object
 */
export async function parseFile(documentation: Documentation, opt: ParseOptions) {
	const source = await read(opt.filePath, {
		encoding: 'utf-8'
	})
	return parseSource(documentation, source, opt)
}

/**
 * parses the source and returns the doc
 * @param {string} source code whose documentation is parsed
 * @param {string} filePath path of the current file against whom to resolve the mixins
 * @returns {object} documentation object
 */
export async function parseSource(documentation: Documentation, source: string, opt: ParseOptions) {
	// if the parsed component is the result of a mixin or an extends
	documentation.setOrigin(opt)
	const singleFileComponent = /\.vue$/i.test(path.extname(opt.filePath))
	let scriptSource: string | undefined

	if (source === '') {
		throw new Error(ERROR_EMPTY_DOCUMENT)
	}

	if (singleFileComponent) {
		// use padding so that errors are displayed at the correct line
		const parts = cacher(() => parseComponent(source, { pad: 'line' }), source)

		if (parts.customBlocks) {
			const docsBlocks = parts.customBlocks
				.filter(block => block.type === 'docs' && block.content && block.content.length)
				.map(block => block.content.trim())

			if (docsBlocks.length) {
				documentation.setDocsBlocks(docsBlocks)
			}
		}

		// get slots and props from template
		if (parts.template) {
			const extTemplSrc: string =
				parts && parts.template && parts.template.attrs ? parts.template.attrs.src : ''
			const extTemplSource =
				extTemplSrc && extTemplSrc.length
					? await read(path.resolve(path.dirname(opt.filePath), extTemplSrc), {
							encoding: 'utf-8'
					  })
					: ''
			if (extTemplSource.length) {
				parts.template.content = extTemplSource
			}
			const addTemplateHandlers: TemplateHandler[] = opt.addTemplateHandlers || []
			parseTemplate(
				parts.template,
				documentation,
				[...templateHandlers, ...addTemplateHandlers],
				opt.filePath
			)
		}

		const extSrc: string = parts && parts.script && parts.script.attrs ? parts.script.attrs.src : ''
		const extSource =
			extSrc && extSrc.length
				? await read(path.resolve(path.dirname(opt.filePath), extSrc), {
						encoding: 'utf-8'
				  })
				: ''

		scriptSource = extSource.length ? extSource : parts.script ? parts.script.content : undefined
		opt.lang =
			(parts.script && parts.script.attrs && parts.script.attrs.lang === 'ts') ||
			/\.tsx?$/i.test(extSrc)
				? 'ts'
				: 'js'
	} else {
		scriptSource = source
		opt.lang = /\.tsx?$/i.test(path.extname(opt.filePath)) ? 'ts' : 'js'
	}

	if (scriptSource) {
		// if jsx option is not mentionned, parse jsx in components
		opt.jsx = opt.jsx === undefined ? true : opt.jsx

		const addScriptHandlers: ScriptHandler[] = opt.addScriptHandlers || []
		if (scriptSource) {
			await parseScript(
				scriptSource,
				documentation,
				preHandlers,
				[...scriptHandlers, ...addScriptHandlers],
				opt
			)
		}
	}

	if (!documentation.get('displayName')) {
		// a component should always have a display name
		documentation.set('displayName', path.basename(opt.filePath).replace(/\.\w+$/, ''))
	}
}
