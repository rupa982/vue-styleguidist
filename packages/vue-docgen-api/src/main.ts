import Documentation, {
	ComponentDoc,
	BlockTag,
	PropDescriptor,
	SlotDescriptor,
	EventDescriptor,
	MethodDescriptor,
	Param,
	Tag,
	ParamTag
} from './Documentation'
import { DocGenOptions, parseFile, ParseOptions, parseSource as parseSourceLocal } from './parse'

export { TemplateParserOptions } from './parse-template'
export { ScriptHandler, TemplateHandler } from './parse'
export {
	ComponentDoc,
	DocGenOptions,
	ParseOptions,
	Documentation,
	BlockTag,
	PropDescriptor,
	SlotDescriptor,
	EventDescriptor,
	MethodDescriptor,
	Param,
	Tag,
	ParamTag
}

/**
 * Parse the components at filePath and return props, public methods, events and slots
 * @param filePath absolute path of the parsed file
 * @param opts
 */
export async function parse(
	filePath: string,
	opts?: DocGenOptions | { [alias: string]: string }
): Promise<ComponentDoc> {
	return parsePrimitive(async (doc, options) => await parseFile(doc, options), filePath, opts)
}

/**
 * Parse the `source` assuming that it is located at `filePath` and return props, public methods, events and slots
 * @param filePath absolute path of the parsed file
 * @param opts
 */
export async function parseSource(
	source: string,
	filePath: string,
	opts?: DocGenOptions | { [alias: string]: string }
): Promise<ComponentDoc> {
	return parsePrimitive(
		async (doc, options) => await parseSourceLocal(doc, source, options),
		filePath,
		opts
	)
}

function isOptionsObject(opts: any): opts is DocGenOptions {
	return !!opts && (!!opts.alias || opts.jsx !== undefined || !!opts.addScriptHandlers || !!opts.addTemplateHandlers)
}

async function parsePrimitive(
	createDoc: (doc: Documentation, opts: ParseOptions) => Promise<void>,
	filePath: string,
	opts?: DocGenOptions | { [alias: string]: string }
): Promise<ComponentDoc> {
	const doc = new Documentation()
	const options: ParseOptions = isOptionsObject(opts)
		? { ...opts, filePath }
		: { filePath, alias: opts }
	await createDoc(doc, options)
	return doc.toObject()
}
