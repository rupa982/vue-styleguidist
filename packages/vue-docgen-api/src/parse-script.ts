import { ParserPlugin } from '@babel/parser'
import * as bt from '@babel/types'
import { NodePath } from 'ast-types'
import recast from 'recast'
import Map from 'ts-map'
import buildParser from './babel-parser'
import Documentation from './Documentation'
import { ParseOptions } from './parse'
import cacher from './utils/cacher'
import resolveExportedComponent from './utils/resolveExportedComponent'

const ERROR_MISSING_DEFINITION = 'No suitable component definition found'

export type Handler = (
	doc: Documentation,
	componentDefinition: NodePath,
	ast: bt.File,
	opt: ParseOptions
) => Promise<void>

export default async function parseScript(
	source: string,
	documentation: Documentation,
	preHandlers: Handler[],
	handlers: Handler[],
	options: ParseOptions
) {
	const plugins: ParserPlugin[] = options.lang === 'ts' ? ['typescript'] : ['flow']
	if (options.jsx) {
		plugins.push('jsx')
	}

	const ast = cacher(() => recast.parse(source, { parser: buildParser({ plugins }) }), source)
	if (!ast) {
		throw new Error(`${ERROR_MISSING_DEFINITION} on "${options.filePath}"`)
	}

	const componentDefinitions = resolveExportedComponent(ast)

	if (componentDefinitions.size === 0) {
		throw new Error(`${ERROR_MISSING_DEFINITION} on "${options.filePath}"`)
	}

	await executeHandlers(preHandlers, handlers, componentDefinitions, documentation, ast, options)
}

async function executeHandlers(
	preHandlers: Handler[],
	localHandlers: Handler[],
	componentDefinitions: Map<string, NodePath>,
	documentation: Documentation,
	ast: bt.File,
	opt: ParseOptions
) {
	return await Promise.all(
		componentDefinitions.keys().map(async name => {
			const compDef = componentDefinitions.get(name)
			if (compDef && name && (!opt.nameFilter || opt.nameFilter.indexOf(name) > -1)) {
				// execute all handlers in order as order matters
				await preHandlers.reduce(async (_, handler) => {
					await _
					return await handler(documentation, compDef, ast, opt)
				}, Promise.resolve())
				await Promise.all(
					localHandlers.map(async handler => await handler(documentation, compDef, ast, opt))
				)
			}
		})
	)
}
