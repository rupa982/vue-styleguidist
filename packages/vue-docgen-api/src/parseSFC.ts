import { parseComponent } from 'vue-template-compiler'
import * as path from 'path'
import { readFile } from 'fs'
import { promisify } from 'util'
import templateHandlers from './template-handlers'
import cacher from './utils/cacher'
import parseTemplate, { Handler as TemplateHandler } from './parse-template'
import Documentation from './Documentation'
import { ParseOptions } from './parse'
import parseScript from './parse-script'
import defaultScriptHandlers, { preHandlers } from './script-handlers'

const read = promisify(readFile)

export default async function parseSFC(
	initialDoc: Documentation | undefined,
	source: string,
	opt: ParseOptions
): Promise<Documentation[]> {
	let documentation = initialDoc

	// use padding so that errors are displayed at the correct line
	const parts = cacher(() => parseComponent(source, { pad: 'line' }), source)

	// get slots and props from template
	if (parts.template) {
		const extTemplSrc: string =
			parts && parts.template && parts.template.attrs ? parts.template.attrs.src : ''

		const extTemplSource =
			extTemplSrc && extTemplSrc.length
				? await read(path.resolve(path.dirname(opt.filePath), extTemplSrc), {
						encoding: 'utf-8'
				  })
				: ''

		if (extTemplSource.length) {
			parts.template.content = extTemplSource
		}
		const addTemplateHandlers: TemplateHandler[] = opt.addTemplateHandlers || []

		documentation = initialDoc || new Documentation(opt.filePath)

		parseTemplate(
			parts.template,
			documentation,
			[...templateHandlers, ...addTemplateHandlers],
			opt.filePath
		)
	}

	const extSrc: string = parts && parts.script && parts.script.attrs ? parts.script.attrs.src : ''
	const extSource =
		extSrc && extSrc.length
			? await read(path.resolve(path.dirname(opt.filePath), extSrc), {
					encoding: 'utf-8'
			  })
			: ''

	let scriptSource = extSource.length ? extSource : parts.script ? parts.script.content : undefined
	opt.lang =
		(parts.script && parts.script.attrs && parts.script.attrs.lang === 'ts') ||
		/\.tsx?$/i.test(extSrc)
			? 'ts'
			: 'js'

	if (parts.customBlocks) {
		documentation = documentation || new Documentation(opt.filePath)

		const docsBlocks = parts.customBlocks
			.filter(block => block.type === 'docs' && block.content && block.content.length)
			.map(block => block.content.trim())

		if (docsBlocks.length) {
			documentation.setDocsBlocks(docsBlocks)
		}
	}

	const scriptHandlers = opt.scriptHandlers || [
		...defaultScriptHandlers,
		...(opt.addScriptHandlers || [])
	]

	const docs: Documentation[] = scriptSource
		? (await parseScript(
				scriptSource,
				opt.scriptPreHandlers || preHandlers,
				scriptHandlers,
				opt,
				documentation,
				initialDoc !== undefined
		  )) || []
		: // if there is only a template return the template's doc
		  documentation
			? [documentation]
			: []

	if (documentation && !documentation.get('displayName')) {
		// a component should always have a display name
		documentation.set('displayName', path.basename(opt.filePath).replace(/\.\w+$/, ''))
	}

	return docs
}
