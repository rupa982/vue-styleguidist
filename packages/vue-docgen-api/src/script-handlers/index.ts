import classDisplayNameHandler from './classDisplayNameHandler'
import classMethodHandler from './classMethodHandler'
import classPropHandler from './classPropHandler'
import componentHandler from './componentHandler'
import displayNameHandler from './displayNameHandler'
import eventHandler from './eventHandler'
import extendsHandler from './extendsHandler'
import methodHandler from './methodHandler'
import mixinsHandler from './mixinsHandler'
import propHandler from './propHandler'
import slotHandler from './slotHandler'
import { Handler } from '../parse-script'

export {
	classDisplayNameHandler,
	classMethodHandler,
	classPropHandler,
	componentHandler,
	displayNameHandler,
	eventHandler,
	extendsHandler,
	methodHandler,
	mixinsHandler,
	propHandler,
	slotHandler
}

const defaultHandlers: Handler[] = [
	displayNameHandler,
	componentHandler,
	methodHandler,
	propHandler,
	eventHandler,
	slotHandler,
	classDisplayNameHandler,
	classMethodHandler,
	classPropHandler
]

export const preHandlers: Handler[] = [
	// have to be first if they can be overridden
	extendsHandler,
	// have to be second as they can be overridden too
	mixinsHandler
]

export default defaultHandlers
