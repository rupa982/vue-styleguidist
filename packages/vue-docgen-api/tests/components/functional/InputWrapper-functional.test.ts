import * as path from 'path'
import { ComponentDoc } from '../../../src/Documentation'
import { parse } from '../../../src/main'

const button = path.join(__dirname, './InputWrapper.vue')
let docButton: ComponentDoc

describe('tests button functional', () => {
	beforeEach(done => {
		docButton = parse(button)
		done()
	})

	it('should extract props from template if functional', () => {
		expect(docButton.props).toMatchObject({
			error: { type: { name: 'boolean' } },
			label: { type: { name: 'string' } },
			'v-model': { type: { name: 'string' } }
		})
	})

	it('should not return the value props as it is v-model', () => {
		expect(docButton.props).not.toMatchObject({
			value: { type: { name: 'string' } }
		})
	})

	it('should match the snapshot', () => {
		expect(docButton).toMatchSnapshot()
	})
})
