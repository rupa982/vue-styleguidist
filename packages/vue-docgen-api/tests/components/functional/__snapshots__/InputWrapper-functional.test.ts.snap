// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests button functional should match the snapshot 1`] = `
Object {
  "description": "Wrapper for input elements",
  "displayName": "InputWrapper",
  "docsBlocks": Array [
    "\`\` \`jsx
<input-wrapper label=\\"Label\\" value=\\"–\\" id=\\"sdf\\">
  …
</input-wrapper>
\`\` \`",
  ],
  "events": undefined,
  "functional": true,
  "methods": undefined,
  "props": Array [
    Object {
      "defaultValue": Object {
        "func": false,
        "value": "false",
      },
      "description": "Should display error?",
      "name": "error",
      "required": undefined,
      "tags": Object {},
      "type": Object {
        "name": "boolean",
      },
    },
    Object {
      "defaultValue": Object {
        "func": false,
        "value": "false",
      },
      "description": "Should the input be focused?",
      "name": "focused",
      "required": undefined,
      "tags": Object {},
      "type": Object {
        "name": "boolean",
      },
    },
    Object {
      "defaultValue": Object {
        "func": false,
        "value": "''",
      },
      "description": "Prefix icon for input field",
      "name": "icon",
      "required": undefined,
      "tags": Object {},
      "type": Object {
        "name": "string",
      },
    },
    Object {
      "defaultValue": Object {
        "func": false,
        "value": "false",
      },
      "description": "Should the input be disabled?",
      "name": "disabled",
      "required": undefined,
      "tags": Object {},
      "type": Object {
        "name": "boolean",
      },
    },
    Object {
      "description": "",
      "name": "value",
      "tags": Object {},
      "type": Object {
        "name": "undefined",
      },
    },
    Object {
      "defaultValue": Object {
        "func": false,
        "value": "false",
      },
      "description": "True if input label should always be \\"up there\\"",
      "name": "labelAlwaysFloating",
      "required": undefined,
      "tags": Object {},
      "type": Object {
        "name": "boolean",
      },
    },
    Object {
      "description": "Prefix string for input field",
      "name": "prefix",
      "tags": Object {},
      "type": Object {
        "name": "string",
      },
    },
    Object {
      "description": "Input ID",
      "name": "id",
      "required": true,
      "tags": Object {},
      "type": Object {
        "name": "string|number",
      },
    },
    Object {
      "description": "String for label text",
      "name": "label",
      "required": true,
      "tags": Object {},
      "type": Object {
        "name": "string",
      },
    },
    Object {
      "defaultValue": Object {
        "func": false,
        "value": "''",
      },
      "description": "String with relevant error message",
      "name": "errorMessage",
      "required": undefined,
      "tags": Object {},
      "type": Object {
        "name": "string",
      },
    },
    Object {
      "description": "Input value",
      "name": "v-model",
      "required": true,
      "tags": Object {
        "model": Array [
          Object {
            "description": true,
            "title": "model",
          },
        ],
      },
      "type": Object {
        "name": "string",
      },
    },
  ],
  "slots": Array [
    Object {
      "bindings": Object {},
      "description": "",
      "name": "default",
    },
  ],
  "tags": Object {},
}
`;
